pythreejs/nbextension/pythreejs.js

```diff
@@ -442,6 +444,50 @@ define(["nbextensions/widgets/widgets/js/widget", "nbextensions/widgets/widgets/
     });
     register['FlyControlsView'] = FlyControlsView;
 
+    var TrackballControlsView = ThreeView.extend({
+        new_properties: function() {
+            ThreeView.prototype.new_properties.call(this);
+            this.array_properties.push('target');
+        },
+
+        render: function() {
+            var that = this;
+            return utils.resolve_promises_dict(this.model.get('controlling').views).then(function(views) {
+                // get the view that is tied to the same renderer
+                that.controlled_view = _.find(views, function(o) {
+                    return o.options.renderer_id === that.options.renderer_id
+                }, that);
+                that.obj = new THREE.TrackballControls(that.controlled_view.obj, that.options.dom);
+                that.register_object_parameters();
+                that.obj.noKeys = true; // turn off keyboard navigation
+                that.options.register_update(that.obj.update, that.obj);
+                that.obj.addEventListener('change', that.options.render_frame);
+                that.obj.addEventListener('start', that.options.start_update_loop);
+                that.obj.addEventListener('end', that.options.end_update_loop);
+                that.obj.addEventListener('end', function() { that.update_controlled(); });
+                // resize again because domElement.getBoundingClientRect() returned all zeros when it's first called
```

Comment:
Agree and agree.

I have tried to attach the resize to TrackballControls "end" event, but it only get the valid rect *after* completion of the first mouse action (after dom mouseup).  I also tried to attach to "start", but at that time the rect isn't valid.  I think to fix the touch interface I can try to hook to a dom touch event.

It is nice to change TrackballControls state as window resizes.  I didn't do it because I don't know how to test it.  So far I don't know how to change the canvas size in ipywidget after it is instantiated.  Do you know how to do it?  Right now in my test environment, no matter how the browser window is resized, the widget/canvas remains the same dimension, and then the control state would never change.